src/MongoDB.Driver.Core/Core/Connections/CommandEventHelper.cs

```diff
@@ -298,7 +309,7 @@ private void ProcessCommandRequestMessage(CommandRequestMessage originalMessage,
                         OperationId = operationId,
                         Stopwatch = stopwatch,
                         QueryNamespace = new CollectionNamespace(databaseNamespace, "$cmd"),
-                        ExpectedResponseType = ExpectedResponseType.Command
+                        ExpectedResponseType = decodedMessage.MoreToCome ? ExpectedResponseType.None : ExpectedResponseType.Command
```

Comment:
This is sufficient for the existing code to generate a synthetic CommandSucceededEvent for unacknowledged writes.